lib/roby/interface/client.rb

```diff
@@ -336,8 +337,18 @@ def async_call(path, m, *args, &block)
                     end
                 end
                 io.write_packet([path, m, *args])
-                pending_async_calls << { block: block, path: path,
-                                         m: m, args: args }
+                pending_async_calls << { block: block, path: path, m: m, args: args }
+                pending_async_calls.last.freeze
```

Comment:
This is fine ... but for your information `#freeze` modifies an object in-place (so, you're freezing the object that is stored within `pending_async_calls`)

Reply:
Yep, definitely. I thoght it made sense because the call is immutable anyway. I do however believe that `#freeze!` would be more consistent with ruby's own idiom (map!, sort_by!, merge!, upcase!, etc...)

Reply:
Well ... the `!` is not used as a mark for self-modification (I would prefer, personally), but to indicate a "dangerous" version of an otherwise non-dangerous method (i.e. map vs map!, sort vs. sort!, ...)

Reply:
My understanding was that "dangerous" in this context meant "modifies the internal state of the receiver"